src/secret_sharing/gf2_array.rs

```diff
@@ -0,0 +1,221 @@
+use std::{
+    fmt::Debug,
+    ops::{Add, AddAssign, Mul, Neg, Sub, SubAssign},
+};
+
+use bitvec::prelude::{BitSlice, Lsb0};
+use generic_array::GenericArray;
+use typenum::U1;
+
+use crate::{
+    ff::{Gf2, Serializable},
+    secret_sharing::{SharedValue, SharedValueArray},
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Gf2Array<const N: usize>([usize; N]);
```

Comment:
I think it ought to be possible to use BAs rather than have a separate Gf2Array type. I didn't put particular thought into using `[usize; N]` here, I probably did it because it was closest to the `[T; N]` used by StdArray. Bitslices ought to do.

Reply:
Do you plan to store a single bit in each usize then? or do you pack them and split them in usize chunks? We need access each bit individually but at the same time want to have a compact representation in memory and network messages.